shotover-proxy/tests/helpers/cassandra.rs

```diff
@@ -157,21 +392,81 @@ impl CassandraConnection {
+                    builder = builder.add_query(query, query_values!(value));
+                }
+
+                let batch = builder.build().unwrap();
+
+                futures::executor::block_on(session.batch(batch)).unwrap();
+            }
         }
     }
 
     #[allow(unused)]
-    pub fn execute_batch_expect_err(&self, batch: &Batch) -> CassandraError {
+    pub fn execute_batch_expect_err(&self, queries: Vec<(String, i32)>) -> String {
         match self {
-            CassandraConnection::Datastax { session, .. } => {
-                session.execute_batch(batch).wait().unwrap_err()
+            #[cfg(feature = "cassandra-cpp-driver-tests")]
+            Self::Datastax { session, .. } => {
+                let mut batch = Batch::new(BatchType::LOGGED);
+                for (query, value) in queries {
+                    batch.add_statement(stmt!(query.as_str()).bind_int32(0, value).unwrap());
+                }
+                let error = session.execute_batch(&batch).wait().unwrap_err();
+                if let ErrorKind::CassErrorResult(_, msg, ..) = error.0 {
+                    return msg;
+                }
+
+                panic!("Did not get an error result for {batch:?}");
+            }
+            Self::CdrsTokio { .. } => todo!(),
+        }
+    }
+
+    fn process_cdrs_response(response: Envelope) -> Vec<Vec<ResultValue>> {
+        let version = response.version;
+        let response_body = response.response_body().unwrap();
+
+        match response_body {
+            ResponseBody::Error(err) => {
+                panic!("CQL query Failed with: {err:?}")
             }
+            ResponseBody::Result(res_result_body) => match res_result_body {
+                ResResultBody::Rows(rows) => {
+                    let mut result_values = vec![];
+
+                    for row in &rows.rows_content {
+                        let mut row_result_values = vec![];
+                        for (i, col_spec) in rows.metadata.col_specs.iter().enumerate() {
+                            let wrapper = wrapper_fn(&col_spec.col_type.id);
+                            let value = ResultValue::new_from_cdrs(
+                                wrapper(&row[i], &col_spec.col_type, version).unwrap(),
+                                version,
+                            );
+
+                            row_result_values.push(value);
+                        }
+                        result_values.push(row_result_values);
+                    }
+
+                    result_values
+                }
+                ResResultBody::Prepared(_) => todo!(),
+                ResResultBody::SchemaChange(_) => vec![],
+                ResResultBody::SetKeyspace(_) => vec![],
+                ResResultBody::Void => vec![],
+            },
+            _ => todo!(),
         }
     }
 }
 
 #[derive(Debug, Clone, PartialOrd, Eq, Ord)]
 pub enum ResultValue {
+    #[cfg(feature = "cassandra-cpp-driver-tests")]
```

Comment:
why is this feature flagged out?

Reply:
I definitely had some reason 😂, but not needed anymore. I'll remove that.

Reply:
Because only the cassandra-cpp driver uses it. I'm inclined to remove it but another driver we add might use it.

Reply:
I took a look at the protocol spec.
It is a bit of a mess but what I'm reading from it is that varchar and text share the same id: `0x000D` so theres no way to differentiate them at the protocol level.
So lets completely remove the text variant and just map anything that comes up as text into Varchar. 

IMO text is a nicer name and we should just use that instead of varchar everywhere but thats for a future cleanup PR.